### ISSUE TITLE
`modulate_with_cenocc` does not behave as expected

### ISSUE THREAD

**Issue by mclaughlin6464** [astropy/halotools#577]
Currently, in the `Zheng07Sats` model, if `modulate_with_cenocc` is set to true a new `Zheng07Cens` object  is created. This object is the one used to "modulate" the satellite mean, not the one in the larger `HODModelFactory`. So, if the user passes in a different central model or (as in my case) subclasses `Zheng07*` with new parameters, those will be overwritten and not applied correctly. 

I propose that, in lieu of using a boolean variable, the user should pass in the central occupation model they're using to the satellite model if they want to modulate. This shouldn't be challenging, as the user has likely created one anyway, and will guarantee that changes made to the object will be appropriately enacted in the modulation. 

I'm going to make a pull request to implement this change. However, I'm sure this will create issues with people already using the code, so should probably be discussed.

**Comment by aphearin** [comment]
@mclaughlin6464 - good catch on this. I went back and forth on this exact design choice quite a bit before deciding to make the hard-coded choice that any user of Zheng07Sats who wants to use a different central model than Zheng07Cens should just write a different Zheng07Sats. My logic at the time of that decision was that any user using doing modeling this sophisticated would probably be comfortable writing their own satellite model in addition to their own central model. Since at the time of that design choice I was myself the only Halotools user, I just decided to move on and wait until some future power-user came along and actually needed the more generalized behavior. Looks like that time has now arrived.

Here is my worry: if you add this new feature you propose, then the Zheng07Sats model may get intimidatingly complicated to a new user, and this is supposed to be literally the simplest realistic HOD there is. However, I do completely see your point, and I can certainly be convinced that your proposed change would not be such a major design change after all. 

At this point, you are probably by now as much if not more of an expert on the internals of that class as I am, so please do me a favor and think that over some and let me know your opinion. 

I'm going to tag other known users of this class, and also those who have just commented on Zheng07-related default behavior, to see whether they have opinions on this as well. Anyone tagged on the following list should not feel obligated to comment, but if you *do* have an opinion, please speak up within the next week. 
cc @manodeep, @surhudm, @mjvakili, @changhoonhahn, @andrew-zentner

**Comment by aphearin** [comment]
Also @mclaughlin6464 - can you comment on how you discovered this "feature"? Was it through mysterious/unexpected buggy behavior? Or did you know this was going to happen once you decided to switch central occupation models? I ask for purposes of possibly improving documentation if necessary.

**Comment by mclaughlin6464** [comment]
@aphearin I see your point entirely. I do not know what the average use case is. However, the way I plan to implement it should not affect users who aren't using `modulate_with_cenocc`, and it won't complicate things for those who are TOO MUCH. The last thing I want to do, though, is make legacy issues for everyone! There are a few possible work arounds for my own use case, but this seemed like the most natural solution. 

I discovered this issue because I'm working on a subclass of `Zheng07*` with new default parameters and some slight modifications. I made some plots that didn't look right given our parameters. I found that my parameters were being overwritten with the defaults. When I looked to see why, it was because Zheng07Sats carries around the defaults from Zheng07Cens, and just overwrote mine from my own central model.
